api/v1/configurationpolicy_types.go

```diff
@@ -65,10 +65,15 @@ type Condition struct {
 	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
 }
 
-// Target defines the list of namespaces to include/exclude
-type Target struct {
+// NamespaceSelector defines the list of namespaces to include/exclude
+type NamespaceSelector struct {
+	// +kubebuilder:validation:MinItems=1
 	Include []NonEmptyString `json:"include,omitempty"`
+	// +kubebuilder:validation:MinItems=1
 	Exclude []NonEmptyString `json:"exclude,omitempty"`
+
+	MatchLabels      *map[string]string                 `json:"matchLabels,omitempty"`
+	MatchExpressions *[]metav1.LabelSelectorRequirement `json:"matchExpressions,omitempty"`
```

Comment:
Setting `// +kubebuilder:validation:MinItems=1` would be nice for this.

Reply:
True, but technically providing it empty returns all namespaces, so I don't think we necessarily want to prevent users from providing an empty MatchExpressions?

Reply:
Fair point!